scrape/target.go

```diff
@@ -366,13 +366,19 @@ type bucketLimitAppender struct {
 
 func (app *bucketLimitAppender) AppendHistogram(ref storage.SeriesRef, lset labels.Labels, t int64, h *histogram.Histogram, fh *histogram.FloatHistogram) (storage.SeriesRef, error) {
 	if h != nil {
-		if len(h.PositiveBuckets)+len(h.NegativeBuckets) > app.limit {
-			return 0, errBucketLimit
+		for len(h.PositiveBuckets)+len(h.NegativeBuckets) > app.limit {
+			if h.Schema == -4 {
```

Comment:
nit: would be nice to have an exported constant for this, I had to redefine in [Mimir](https://github.com/grafana/mimir/blob/6eb5f34818b32c91d113101e7a6cb16d4842f3b0/pkg/mimirpb/custom.go#L63) as well. A follow PR is ok I think.

Reply:
Ok, but I'm not sure where to define, in prometheus or client_golang? @beorn7 do you have any suggestions?

Reply:
I'm not so sure we should do it cross-repo. It's currently "part of the (not yet written) standard", but it might also get updated if need be. And then it might very well be that Prometheus itself supports a wider range of schemas than client_golang (e.g. if it is only needed to read OTel properly, while client_golang isn't concerned with reading OTel).

As a first step, there could be a constant in prometheus/prometheus/model/histogram.

But as said, doesn't have to be this PR.